packages/data-broker/src/services/dhis/DhisService.ts

```diff
@@ -295,28 +260,32 @@ export class DhisService extends Service {
   private deleteEvent = async (api: DhisApi, data: DeleteEventData) =>
     api.deleteEvent(data.dhisReference);
 
-  public async pull(
-    dataSources: DataElement[],
-    type: 'dataElement',
-    options: PullAnalyticsOptions,
-  ): Promise<AnalyticResults>;
-  public async pull(
-    dataSources: DataGroup[],
-    type: 'dataGroup',
-    options: PullEventsOptions,
-  ): Promise<EventResults>;
-  public async pull(dataSources: DataSource[], type: DataSourceType, options: PullOptions) {
+  public async pullAnalytics(dataElements: DataElement[], options: PullAnalyticsOptions) {
+    const { dataServiceMapping } = options;
+
+    const dhisDataElements = dataElements.filter(
+      ds => dataServiceMapping.mappingForDataSource(ds)?.service_type === 'dhis',
+    );
+    const apis = await getApisForDataSources(this.models, dhisDataElements, dataServiceMapping);
+
+    return this.analyticsPuller.pull.bind(this)(apis, dhisDataElements, options);
+  }
+
+  public async pullEvents(dataGroups: DataGroup[], options: PullEventsOptions) {
     const { dataServiceMapping, useDeprecatedApi = false } = options;
 
-    const dhisDataSources = dataSources.filter(
+    const dhisDataGroups = dataGroups.filter(
       ds => dataServiceMapping.mappingForDataSource(ds)?.service_type === 'dhis',
     );
+    const apis = await getApisForDataSources(this.models, dhisDataGroups, dataServiceMapping);
 
-    const apis = await getApisForDataSources(this.models, dhisDataSources, dataServiceMapping);
-    const pullerKey = `${type}${useDeprecatedApi ? '_deprecated' : ''}`;
+    return useDeprecatedApi
+      ? this.deprecatedEventsPuller.pull.bind(this)(apis, dhisDataGroups, options)
```

Comment:
I wonder if the deprecated event API can be removed now :thinking:

Reply:
I don't think we can, sadly!